strongbox-security/strongbox-security-api/src/main/java/org/carlspring/strongbox/configuration/StrongboxDelegatingPasswordEncoder.java

```diff
@@ -0,0 +1,88 @@
+package org.carlspring.strongbox.configuration;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.security.crypto.factory.PasswordEncoderFactories;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Base64Utils;
+
+/**
+ * This class encodes given raw String or Base64 encoded string.
+ * If given string is Base64 Encoded it will attempt to decode the password and finally delegate it to the password encoder.
+ *
+ * @author mbharti
+ * @date 07/10/20
+ */
+@Component
+public class StrongboxDelegatingPasswordEncoder
+        implements PasswordEncoder
+{
+
+    private PasswordEncoder passwordEncoder;
+
+    @PostConstruct
+    private void init()
+    {
+        passwordEncoder = PasswordEncoderFactories.createDelegatingPasswordEncoder();
+    }
```

Comment:
I think this should be a constructor - there's no point in having a `@PosConstruct`. It will also produce a `NPE` when you try to instantiate via `new StrongboxDelegatingPasswordEncoder()` because the `private PasswordEncoder passwordEncoder` will be null.